File: app/src/main/java/com/lawrence/superheroapp/rest/RetrofitClient.java
package com.lawrence.superheroapp.rest;


import android.os.Build;

import java.io.IOException;
import java.util.Locale;

import okhttp3.Cookie;
import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {

    private static String BASE_URL = "https://superheroapi.com//";
    private static String TOKEN = "105581651029925";

    private static HttpLoggingInterceptor logger =
            new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY);

    private static OkHttpClient.Builder okHttp =
            new OkHttpClient.Builder()
                    .addInterceptor(new Interceptor() {
                        @Override
                        public Response intercept(Chain chain) throws IOException {
                            Request request = chain.request();

                            request = request.newBuilder()
                                    .addHeader("Accept-Language", Locale.getDefault().getLanguage())
                                    .addHeader("access-token", "105581651029925")
                                    .build();
                            return chain.proceed(request);
                        }
                    })
                    .addInterceptor(logger);

    private static Retrofit.Builder builder =
            new Retrofit.Builder().baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(okHttp.build());
    private static Retrofit retrofit = builder.build();

    public static <S> S buildService(Class<S> serviceType) {
        return retrofit.create(serviceType);


    }

}

File: app/src/main/java/com/lawrence/superheroapp/rest/RetrofitClient.kt
package com.lawrence.superheroapp.rest

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.*

object RetrofitClient {
    private const val BASE_URL = "https://superheroapi.com//"
    private val logger = HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY)
    private val okHttp = OkHttpClient.Builder()
            .addInterceptor { chain ->
                var request = chain.request()
                request = request.newBuilder()
                        .addHeader("Accept-Language", Locale.getDefault().language)
                        .addHeader("access-token", "105581651029925")
                        .build()
                chain.proceed(request)
            }
            .addInterceptor(logger)
    private val builder = Retrofit.Builder().baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .client(okHttp.build())
    private val retrofit = builder.build()
    @JvmStatic
    fun <S> buildService(serviceType: Class<S>?): S {
        return retrofit.create(serviceType)
    }
}